osu.Game.Rulesets.Taiko/Audio/TaikoHitSampleInfo.cs

```diff
@@ -0,0 +1,108 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System;
+using System.Collections.Generic;
+using osu.Game.Audio;
+using osu.Game.Beatmaps.ControlPoints;
+using osu.Game.Utils;
+
+namespace osu.Game.Rulesets.Taiko.Audio
+{
+    public class TaikoHitSampleInfo : HitSampleInfo
+    {
+        public const string TAIKO_STRONG_HIT = @"taiko-strong-hit";
+        public const string TAIKO_STRONG_CLAP = @"taiko-strong-clap";
+        public const string TAIKO_STRONG_FLOURISH = @"taiko-strong-flourish";
+
+        public const int SAMPLE_VOLUME_THRESHOLD_HARD = 90;
+        public const int SAMPLE_VOLUME_THRESHOLD_MEDIUM = 60;
+
+        /// <summary>
+        /// A taiko-specific velocity suffix, currently applied automatically based on volume.
+        /// </summary>
+        public readonly string VelocitySuffix;
+
+        public TaikoHitSampleInfo(string name, string bank = SampleControlPoint.DEFAULT_BANK, string? suffix = null, int volume = 0)
+            : base(name, bank, suffix, volume)
+        {
+            VelocitySuffix = getVelocitySuffix(name, volume);
+        }
+
+        private static string getVelocitySuffix(string name, int volume)
+        {
+            switch (name)
+            {
+                case HIT_NORMAL:
+                case HIT_CLAP:
+                {
+                    if (volume >= SAMPLE_VOLUME_THRESHOLD_HARD)
+                        return "-hard";
+
+                    if (volume >= SAMPLE_VOLUME_THRESHOLD_MEDIUM)
+                        return "-medium";
+
+                    return "-soft";
+                }
+
+                default:
+                    return string.Empty;
+            }
+        }
+
+        public override IEnumerable<string> LookupNames
```

Comment:
I will say that the amount of complexity in this here property does scare me a little bit. It seems to be working okay, but the logic seems rather brittle and prone to breakage, and I'm not even sure if it can be tested properly. Would be nice if it were tested, though.